src/parser/SparqlParser.cpp

```diff
@@ -761,11 +763,19 @@ bool SparqlParser::parseFilter(
           escaped = !escaped;  // correctly deal with consecutive backslashes
           continue;
         }
-        if (!escaped) {
-          char c = f._rhs[i];
-          if (regexControlChars.find(c) != string::npos) {
-            isSimple = false;
-          }
+        char c = f._rhs[i];
+        bool isControlChar = regexControlChars.find(c) != string::npos;
+        if (!escaped && isControlChar) {
+          isSimple = false;
+        } else if (escaped && !isControlChar) {
+          const std::string error =
+              "Escaping the character "s + c +
```

Comment:
Why do you have the `std::literals::string_literals::operator""s` only on the first literal?